pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/IdSetAggregationFunction.java

```diff
@@ -225,63 +324,130 @@ public void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResult
+            byte[] bytesValue = bytesValuesSV[i];
+            for (int groupKey : groupKeysArray[i]) {
+              getIdSet(groupByResultHolder, groupKey, DataType.BYTES).add(bytesValue);
+            }
           }
-        }
-        break;
-      default:
-        throw new IllegalStateException("Illegal data type for ID_SET aggregation function: " + storedType);
+          break;
+        default:
+          throw new IllegalStateException("Illegal SV data type for ID_SET aggregation function: " + storedType);
+      }
+    } else {
+      switch (storedType) {
+        case INT:
+          int[][] intValuesMV = blockValSet.getIntValuesMV();
+          for (int i = 0; i < length; i++) {
+            int[] intValues = intValuesMV[i];
+            for (int groupKey : groupKeysArray[i]) {
+              IdSet idSet = getIdSet(groupByResultHolder, groupKey, DataType.INT);
+              for (int intValue : intValues) {
+                idSet.add(intValue);
+              }
+            }
+          }
+          break;
+        case LONG:
+          long[][] longValuesMV = blockValSet.getLongValuesMV();
+          for (int i = 0; i < length; i++) {
+            long[] longValues = longValuesMV[i];
+            for (int groupKey : groupKeysArray[i]) {
+              IdSet idSet = getIdSet(groupByResultHolder, groupKey, DataType.LONG);
+              for (long longValue : longValues) {
+                idSet.add(longValue);
+              }
+            }
+          }
+          break;
+        case FLOAT:
+          float[][] floatValuesMV = blockValSet.getFloatValuesMV();
+          for (int i = 0; i < length; i++) {
+            float[] floatValues = floatValuesMV[i];
+            for (int groupKey : groupKeysArray[i]) {
+              IdSet idSet = getIdSet(groupByResultHolder, groupKey, DataType.FLOAT);
+              for (float floatValue : floatValues) {
+                idSet.add(floatValue);
+              }
+            }
+          }
+          break;
+        case DOUBLE:
+          double[][] doubleValuesMV = blockValSet.getDoubleValuesMV();
+          for (int i = 0; i < length; i++) {
+            double[] doubleValues = doubleValuesMV[i];
+            for (int groupKey : groupKeysArray[i]) {
+              IdSet idSet = getIdSet(groupByResultHolder, groupKey, DataType.DOUBLE);
+              for (double doubleValue : doubleValues) {
+                idSet.add(doubleValue);
+              }
+            }
+          }
+          break;
+        case STRING:
+          String[][] stringValuesMV = blockValSet.getStringValuesMV();
+          for (int i = 0; i < length; i++) {
+            String[] stringValues = stringValuesMV[i];
+            for (int groupKey : groupKeysArray[i]) {
+              IdSet idSet = getIdSet(groupByResultHolder, groupKey, DataType.STRING);
+              for (String stringValue : stringValues) {
+                idSet.add(stringValue);
+              }
+            }
+          }
+          break;
```

Comment:
add the bytes support too?

Reply:
We don't have MV bytes, bytes are supported as SV